### PR TITLE
[Settings] More

### PR DESCRIPTION
**Pull request by JuancaG05**
Implements #3100

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsFragment.kt [hunk 1]
```diff
@@ -204,6 +221,112 @@ class SettingsFragment : PreferenceFragmentCompat() {
...
+        } else {
+            prefMoreCategory?.removePreference(prefRecommend)
+        }
+
+        // Feedback
+        if (settingsViewModel.isFeedbackEnabled()) {
+            prefFeedback?.setOnPreferenceClickListener {
+                val feedbackMail = getString(R.string.mail_feedback)
+                val feedback = "Android v" + BuildConfig.VERSION_NAME + " - " + getString(R.string.prefs_feedback)
+                val intent = Intent(Intent.ACTION_SENDTO)
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsFragment.kt:284]: I can see that we build mail intents more than one time in the app (at least two times in this file). I think that we could unify them as you did when [mocking intents](https://github.com/owncloud/android/blob/settings%2Fandroidx/owncloudApp/src/androidTest/java/com/owncloud/android/utils/IntentsUtil.kt). You could create a new Fragment extension [here](https://github.com/owncloud/android/blob/settings%2Fandroidx/owncloudApp/src/main/java/com/owncloud/android/extensions/FragmentExt.kt) and keep learning Kotlin things 🍻

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsFragment.kt [hunk 2]
```diff
@@ -204,6 +221,112 @@ class SettingsFragment : PreferenceFragmentCompat() {
...
+                true
+            }
+        } else {
+            prefMoreCategory?.removePreference(prefImprint)
+        }
+
+        // About app
+        prefAboutApp?.let {
+            it.title = String.format(getString(R.string.about_android), getString(R.string.app_name))
+            val appVersion = BuildConfig.VERSION_NAME + " " + BuildConfig.BUILD_TYPE + " " + BuildConfig.COMMIT_SHA1
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsFragment.kt:321]: Just a suggestion, nothing relevant. But I prefer this way to concatenate strings in Kotlin. Up to you 👍           
```suggestion
            val appVersion = "${BuildConfig.VERSION_NAME} ${BuildConfig.BUILD_TYPE} ${BuildConfig.COMMIT_SHA1}"

```

File: owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsViewModelTest.kt [hunk 3]
```diff
@@ -333,4 +339,203 @@ class SettingsViewModelTest : ViewModelTest() {
...
+        assertEquals(EXAMPLE_URL, imprintUrl)
+
+        verify(exactly = 1) {
+            contextProvider.getString(R.string.url_imprint)
+        }
+    }
+
+    companion object {
+        private const val EXAMPLE_URL = "URL"
+    }
```
Comment (abelgardep) [inline owncloudApp/src/test/java/com/owncloud/android/presentation/viewmodels/settings/SettingsViewModelTest.kt:539]: You could use `OC_BASE_URL` for this purpose

File: owncloudApp/src/androidTest/java/com/owncloud/android/settings/more/SettingsFragmentMoreSectionTest.kt [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+        assertEquals(PREFERENCE_PRIVACY_POLICY, prefPrivacyPolicy?.key)
+        assertEquals(context.getString(R.string.prefs_privacy_policy), prefPrivacyPolicy?.title)
+        assertEquals(null, prefPrivacyPolicy?.summary)
+        assertTrue(prefPrivacyPolicy?.isVisible == true)
+        assertTrue(prefPrivacyPolicy?.isEnabled == true)
+
+        onView(withText(R.string.prefs_imprint)).check(matches(isDisplayed()))
+        assertEquals(PREFERENCE_IMPRINT, prefImprint?.key)
+        assertEquals(context.getString(R.string.prefs_imprint), prefImprint?.title)
+        assertEquals(null, prefImprint?.summary)
```
Comment (abelgardep) [inline owncloudApp/src/androidTest/java/com/owncloud/android/settings/more/SettingsFragmentMoreSectionTest.kt:202]:
```suggestion
        assertNull(prefImprint?.summary)
```
Same for others `assertEquals(null, ...)`

File: owncloudApp/src/androidTest/java/com/owncloud/android/settings/more/SettingsFragmentMoreSectionTest.kt [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+        private const val PREFERENCE_MORE_CATEGORY = "more_category"
+        private const val PREFERENCE_HELP = "help"
+        private const val PREFERENCE_SYNC_CALENDAR_CONTACTS = "syncCalendarContacts"
+        private const val PREFERENCE_RECOMMEND = "recommend"
+        private const val PREFERENCE_FEEDBACK = "feedback"
+        private const val PREFERENCE_PRIVACY_POLICY = "privacyPolicy"
+        private const val PREFERENCE_IMPRINT = "imprint"
+        private const val PREFERENCE_ABOUT_APP = "about_app"
+    }
+}
```
Comment (abelgardep) [inline owncloudApp/src/androidTest/java/com/owncloud/android/settings/more/SettingsFragmentMoreSectionTest.kt:380]: Try to add an empty line at the end of files 👍

**Review by abelgardep — CHANGES_REQUESTED** [review]
Some suggestions here @JuancaG05. Great job!

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsFragment.kt (hunk 1 above)
Comment (JuancaG05) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsFragment.kt:284]: Finally moved to Activity extension as we discussed, and new issue (https://github.com/owncloud/android/issues/3138) created to adapt the rest of the project

File: owncloudApp/src/androidTest/java/com/owncloud/android/settings/more/SettingsFragmentMoreSectionTest.kt (hunk 5 above)
Comment (JuancaG05) [reply owncloudApp/src/androidTest/java/com/owncloud/android/settings/more/SettingsFragmentMoreSectionTest.kt:380]: My fault 😬

**Review by abelgardep — APPROVED** [review]
Good job @JuancaG05! Let's move it forward!

**Comment by abelgardep** [comment]
QA is not needed for the moment. It will be tested when all sections are migrated. We can merge it into `settings/androidx`

**Pull request merged**